### DISCUSSION TITLE
Writing a filesystem in .NET Core?

### DISCUSSION THREAD

**Discussion by aled** [winfsp/winfsp#484] [Q&A] [answered]
I have successfully compiled the passthrough filesystem (using the included winfsp-msil.dll), in a .NET Framework executable.

Before I start looking at how to make this work in .NET Core, I thought I'd ask if there are any existing efforts to do this, or if there is any advice on the best way to go about it?

(Yes, I know it's not called Core anymore)

Thanks in advance.

**Comment by billziss-gh** [comment, marked as answer]
There has been a recent PR (#451) which added .NET Core support to WinFsp. I have not made a release with this support yet, but I intend to have it added to WinFsp 2023 (when it is released).
